src/test/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/segmentgroup/InvestigationSubjectTest.java

```diff
@@ -35,10 +40,23 @@ void testGetUnstructuredAddress() {
             "ADR++US:FLAT1:12 BROWNBERRIE AVENUE::LEEDS:++LS18 5PN",
             "ignore me"
         ));
-        assertThat(investigationSubject.getAddress())
-            .isPresent()
-            .map(UnstructuredAddress::getValue)
-            .contains("US:FLAT1:12 BROWNBERRIE AVENUE::LEEDS:++LS18 5PN");
+        var investigationSubjectAddress = assertThat(investigationSubject.getAddress()).isPresent();
+
+        var address = investigationSubjectAddress
+            .map(UnstructuredAddress::getAddressLines);
+
+        var expectedValues = List.of("FLAT1", "12 BROWNBERRIE AVENUE", "", "LEEDS", "");
+        for (int i = 0; i < expectedValues.size(); i++) {
+            final int index = i;
+            address.map(x -> x[index]).hasValue(expectedValues.get(i));
+        }
```

Comment:
assertj to the rescue
```suggestion
        investigationSubjectAddress
            .map(UnstructuredAddress::getAddressLines)
            .contains(new String[] {"FLAT1", "12 BROWNBERRIE AVENUE", "", "LEEDS", ""});
```